src/components/availability-panel/drag-drop-context/DraggableComponent.tsx

```diff
@@ -38,6 +37,20 @@ const DraggableComponent: FC<draggableProps> = ({
     title = '',
 }) => {
     const draggableIds = useContext(disableDrag);
+    const [draggableUserImageUrl, setdraggableUserImageUrl] =
+        useState(DUMMY_PROFILE);
+    const draggableUserImageOnError = () =>
+        setdraggableUserImageUrl(DUMMY_PROFILE);
+
+    useEffect(() => {
+        getUserProfileImage(draggableId)
+            .then((res) => {
+                setdraggableUserImageUrl(res);
+            })
+            .catch(() => {
+                setdraggableUserImageUrl(DUMMY_PROFILE);
+            });
```

Comment:
Does this mean that if we have a 100 tasks assigned to the **same** user, then we would make 100 image request calls to the same URL via `getUserProfileImage`? That seems wasteful to me.

Should we think about leveraging a global store with some caching enabled to reduce these unnecessary calls?

Reply:
Also, worth checking if this asset would automatically get cached on the browser? After the first load, would subsequent images be served from the browser cache directly or make a fresh network call?

Reply:
> Also, worth checking if this asset would automatically get cached on the browser? After the first load, would subsequent images be served from the browser cache directly or make a fresh network call?

This will be done and maintained by ReduxToolkit. So, it's not required to be done manually.

Reply:
> Does this mean that if we have a 100 tasks assigned to the **same** user, then we would make 100 image request calls to the same URL via `getUserProfileImage`? That seems wasteful to me.
> 
> Should we think about leveraging a global store with some caching enabled to reduce these unnecessary calls?

@ankushdharkar which approach you are referring to for caching data manually caching using useEffect, or using a tool like Redux Toolkit with RTK Query for automatic caching?